index.d.ts

```diff
@@ -124,7 +124,7 @@ export function getUserId(): Promise<string>;
  * @param IgnoredComponent the component to wrap with HeapIgnore.
  * @param heapIgnoreConfig the HeapIgnore configuration
  */
-export function withHeapIgnore(IgnoredComponent: React.Component, heapIgnoreConfig?: object): React.Component;
+ export function withHeapIgnore<P>(IgnoredComponent: React.JSXElementConstructor<P>, heapIgnoreConfig?: object): React.JSXElementConstructor<P>;
```

Comment:
Things like `Button` inherit `React.Component` but what we pass in is something to create the component not a component itself.

Reply:
Huh, was this a mistake that was sitting there?

Reply:
Yeah.  Typescript is fun. We get some same weirdness in Objective-C when you deal with passing classes around as protocol implementations.  I noticed the errors when bringing HeapIgnore.tsx from the old e2e project.

The definition is:

```
    type JSXElementConstructor<P> =
        | ((props: P) => ReactElement<any, any> | null)
        | (new (props: P) => Component<any, any>);
```

Beyond silencing the errors, this actually improves things in two ways:

1. It also removes errors on element functions like:

    ```tsx
    const Foo = ({children}) => {
      return <>{children}</>;
    };

    const IggyFoo = Heap.withHeapIgnore(Foo);
    ```

2. It makes the returned component accept the same params (`<P>`) as the original type, so you get full type checking on the element.